lib/tools.js

```diff
@@ -330,92 +359,131 @@ function uuid(givenMac, callback) {
+            }
         }
+
+        _objects.setObject('system.meta.uuid', {
+            type: 'meta',
+            common: {
+                name: 'uuid',
+                type: 'uuid'
+            },
+            ts: new Date().getTime(),
+            from: 'system.host.' + getHostName() + '.tools',
+            native: {
+                uuid: _uuid
+            }
+        }, err => {
+            if (err) {
+                console.error('object system.meta.uuid cannot be updated: ' + err);
+                callback();
+            } else {
+                _objects.getObject('system.meta.uuid', (err, obj) => {
+                    if (obj.native.uuid !== _uuid) {
+                        console.error('object system.meta.uuid cannot be updated: write protected');
+                    } else {
+                        console.log('object system.meta.uuid created: ' + _uuid);
+                    }
+
+                    callback(_uuid);
+                });
+            }
+        });
     });
+}
 
-    _objects.getObject('system.meta.uuid', (err, obj) => {
-        if (!err && obj && obj.native && obj.native.uuid) {
-            !--tasks && callback && callback();
-        } else {
-            uuid(__uuid => {
-                _uuid = __uuid;
+function createUuid(_objects, callback) {
+    const promiseCheckPassword = new Promise(resolve =>
+        _objects.getObject('system.user.admin', (err, obj) => {
+            if (err || !obj) {
+                password = password || require('./password');
+
+                // Default Password for user 'admin' is application name in lower case
+                password(getAppName()).hash(null, null, (err, res) => {
+                    err && console.error(err);
+
+                    // Create user here and not in io-package.js because of hash password
+                    _objects.setObject('system.user.admin', {
+                        type: 'user',
+                        common: {
+                            name:      'admin',
+                            password:   res,
+                            dontDelete: true,
+                            enabled:    true
+                        },
+                        ts: new Date().getTime(),
+                        from: 'system.host.' + getHostName() + '.tools',
+                        native: {}
+                    }, () => {
+                        console.log('object system.user.admin created');
+                        resolve();
+                    });
+                });
+            } else {
+                resolve();
+            }
+        })
+    );
+
+    const promiseCheckUuid = new Promise(resolve =>
+        _objects.getObject('system.meta.uuid', (err, obj) => {
+            if (!err && obj && obj.native && obj.native.uuid) {
+                // if COMMON invalid docker uuid
+                if (obj.native.uuid === 'ab265f4a-67f9-a46a-c0b2-61e4b95cefe5') {
```

Comment:
maybe better to plave in a "blacklist array" because now we have the Docker ID already two times in the code